dist/testbed-support/makefile.iotlab.single.inc.mk

```diff
@@ -55,7 +55,7 @@ IOTLAB_USER ?= $(shell cut -f1 -d: $(IOTLAB_AUTH))
 IOTLAB_EXP_ID ?=
 
 # File to use for flashing
-IOTLAB_FLASHFILE ?= $(ELFFILE)
+FLASHFILE ?= $(ELFFILE)
```

Comment:
Why is change required for the fix to work ?
The commit says "useless" without explaining anything or justifying why it is useless.


Not all board have a "protected" programmer configuration that would not set `FLASHFILE` so it would quite easily use another file. The file used to flash in RIOT can be whichever bin/hex/elf, but on IoT-AB it must be the `ELFFILE` )except for `wsn430.
I feel that changing this assumes IoT-LAB can flash with whichever file produced. If it is now the case, please clarify this.

```
git grep 'FLASHFILE ?='
boards/cc2538dk/Makefile.include:FLASHFILE ?= $(BINFILE)
boards/chronos/Makefile.include:FLASHFILE ?= $(HEXFILE)
boards/common/blxxxpill/Makefile.include:  FLASHFILE ?= $(BINFILE)
boards/common/msb-430/Makefile.include:FLASHFILE ?= $(HEXFILE)
boards/common/msba2/Makefile.include:FLASHFILE ?= $(HEXFILE)
boards/common/nrf52/Makefile.include:    FLASHFILE ?= $(HEXFILE)
boards/common/remote/Makefile.include:FLASHFILE ?= $(BINFILE)
boards/common/wsn430/Makefile.include:FLASHFILE ?= $(HEXFILE)
boards/f4vi1/Makefile.include:FLASHFILE ?= $(BINFILE)
boards/lobaro-lorabox/Makefile.include:FLASHFILE ?= $(BINFILE)
boards/mbed_lpc1768/Makefile.include:FLASHFILE ?= $(BINFILE)
boards/native/Makefile.include:FLASHFILE ?= $(ELFFILE)
boards/nz32-sc151/Makefile.include:FLASHFILE ?= $(BINFILE)
boards/opencm904/Makefile.include:FLASHFILE ?= $(BINFILE)
boards/openmote-b/Makefile.include:  FLASHFILE ?= $(HEXFILE)
boards/openmote-cc2538/Makefile.include:  FLASHFILE ?= $(BINFILE)
boards/pic32-clicker/Makefile.include:FLASHFILE ?= $(HEXFILE)
boards/pic32-wifire/Makefile.include:  FLASHFILE ?= $(HEXFILE)
boards/seeeduino_arch-pro/Makefile.include:FLASHFILE ?= $(HEXFILE)
boards/spark-core/Makefile.include:FLASHFILE ?= $(BINFILE)
boards/teensy31/Makefile.include:FLASHFILE ?= $(HEXFILE)
boards/telosb/Makefile.include:FLASHFILE ?= $(HEXFILE)
boards/z1/Makefile.include:FLASHFILE ?= $(HEXFILE)
cpu/esp32/Makefile.include:FLASHFILE ?= $(ELFFILE)
cpu/esp8266/Makefile.include:FLASHFILE ?= $(ELFFILE)
dist/testbed-support/makefile.iotlab.single.inc.mk:IOTLAB_FLASHFILE ?= $(ELFFILE)
makefiles/tools/avrdude.inc.mk:FLASHFILE ?= $(HEXFILE)
makefiles/tools/bossa.inc.mk:FLASHFILE ?= $(BINFILE)
makefiles/tools/dfu.inc.mk:FLASHFILE ?= $(BINFILE)
makefiles/tools/edbg.inc.mk:FLASHFILE ?= $(BINFILE)
makefiles/tools/jlink.inc.mk:FLASHFILE ?= $(BINFILE)
makefiles/tools/openocd.inc.mk:FLASHFILE ?= $(ELFFILE)
makefiles/tools/pic32prog.inc.mk:FLASHFILE ?= $(HEXFILE)
makefiles/tools/pyocd.inc.mk:FLASHFILE ?= $(HEXFILE)
makefiles/tools/uniflash.inc.mk:FLASHFILE ?= $(ELFFILE)
```

Reply:
IoT-LAB only supports ELF files now (except for WSN4330) but soon it will also support bin files.

How about using `FLASHFILE ?= ELFFILE` as default, early (eg before boards Makefile.include are parsed) for all boards except WSN430 ?

If we agree on that, I'll reword the commit message.

Reply:
I adapted the code because I noticed that it was broken with firefly (iot-lab requires ELFFILE but the build system was already setting BINFILE).